rhino/src/main/java/org/mozilla/javascript/typedarrays/NativeDataView.java

```diff
@@ -241,43 +238,65 @@ private static NativeDataView js_constructor(Context cx, Scriptable scope, Objec
 
         NativeArrayBuffer ab = (NativeArrayBuffer) args[0];
 
-        int pos;
-        if (isArg(args, 1)) {
-            double doublePos = ScriptRuntime.toNumber(args[1]);
-            if (Double.isInfinite(doublePos)) {
-                throw ScriptRuntime.rangeError("offset out of range");
-            }
-            pos = ScriptRuntime.toInt32(doublePos);
-        } else {
-            pos = 0;
+        int pos = ScriptRuntime.toIndex(isArg(args, 1) ? args[1] : Undefined.instance);
+
+        if (ab.isDetached()) {
+            throw ScriptRuntime.typeErrorById("msg.arraybuf.detached");
+        }
+
+        int bufferByteLength = ab.getLength();
+        if (pos > bufferByteLength) {
+            throw ScriptRuntime.rangeErrorById("msg.dataview.offset.range");
```

Comment:
I think we should add some more info to this kind of messages, maybe we can check what browsers are doing